src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -1922,8 +1922,18 @@ QSGNode *RenderWindowItem::updatePaintNode(QSGNode *_node,
     // and makeCurrent down below while setting up our own context.
     current->doneCurrent();
 
+    auto surfaceFormat = QSurfaceFormat();
+    surfaceFormat.setMajorVersion(3);
+    surfaceFormat.setMinorVersion(3);
+    surfaceFormat.setProfile(QSurfaceFormat::CoreProfile);
+    
     this->dataPtr->renderThread->context = new QOpenGLContext();
-    this->dataPtr->renderThread->context->setFormat(current->format());
+    // Although it seems unbelievable, we can request another format for a
+    // shared context; it is needed because Qt selects by default a compat
+    // context which is much less likely to provide OpenGL 3.3 (only closed
+    // NVidia drivers). This means there will be mismatch between what is
+    // reported by QSG_INFO=1 and by OGRE.
+    this->dataPtr->renderThread->context->setFormat(surfaceFormat);
```

Comment:
can you wrap the code with a check for `ogre2` since this requirement is engine-dependent?

```
    if (this->RenderUtil()->EngineName() == "ogre2")
    {
      ...
      this->dataPtr->renderThread->context->setFormat(surfaceFormat);
    }
    else
    {
      this->dataPtr->renderThread->context->setFormat(current->format());
    }
```

 I think in the future we should support querying the underlying render engine for the min required opengl version so we don't need to hard code the engine name check.

Reply:
You're right, I fixed that.